### PR TITLE
feat(intent): Security group ingress rule support

### PR DESCRIPTION
**Pull request by robzienert**
Not all-inclusive, but a start with the most common case.

### CODE REVIEW

File: keel-intents/src/main/kotlin/com/netflix/spinnaker/keel/intents/processors/SecurityGroupIntentProcessor.kt [hunk 1]
```diff
@@ -73,21 +73,20 @@ class SecurityGroupIntentProcessor
...
+      // TODO rz - It'd be rad to enumerate what actually changed in the reason if it's a delta, rather than giving
+      // a generic message.
       ConvergeReason.CHANGED.reason
     )
   }
 
   private fun getSecurityGroups(spec: SecurityGroupSpec): Set<SecurityGroup> {
-    // TODO rz - Quite dislike that any new cloudprovider would need to edit keel for declarative support... how to fix?
-    // Maybe this is the sacrifice we need to make to start statically typing all of our models?
     if (spec is AmazonSecurityGroupSpec) {
```
Comment (emjburns) [inline keel-intents/src/main/kotlin/com/netflix/spinnaker/keel/intents/processors/SecurityGroupIntentProcessor.kt]: Any thoughts about how we're gonna handle this for other types of security groups?

File: keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/OrcaIntentLauncher.kt [hunk 2]
```diff
@@ -40,25 +40,32 @@ open class OrcaIntentLauncher
...
-        .also {
-          if (it.orchestrations.isNotEmpty()) {
-            applicationEventPublisher.publishEvent(ConvergenceRequiredEvent(intent))
-          }
-        }.orchestrations.map {
-        log.info("Launching orchestration for intent (kind: ${intent.kind})")
+      val result = intentProcessor(intentProcessors, intent).converge(intent)
+
+      if (result.orchestrations.isEmpty()) {
+        log.info("Not converging state for intent (id: ${intent.id}, reason: ${result.reason})")
```
Comment (emjburns) [inline keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/OrcaIntentLauncher.kt]: Should be structured logs now?

File: keel-intents/src/main/kotlin/com/netflix/spinnaker/keel/intents/processors/SecurityGroupIntentProcessor.kt (hunk 1 above)
Comment (robzienert) [reply keel-intents/src/main/kotlin/com/netflix/spinnaker/keel/intents/processors/SecurityGroupIntentProcessor.kt]: Each cloud provider will need to bring their own implementation at this point, just as they need to bring their own implementation into orca stages.

File: keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/OrcaIntentLauncher.kt (hunk 2 above)
Comment (robzienert) [reply keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/OrcaIntentLauncher.kt]: been updated with the rebase. :)

**Comment by emjburns** [comment]
LGTM

**Pull request merged**